src/components/TopBanner.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash/fp'
+import { div, h } from 'react-hyperscript-helpers'
+import Interactive from 'react-interactive'
+import { icon } from 'src/components/icons'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+
+
+/**
+ * @param {bool} isVisible
+ * @param {bool} [showX=true]
+ * @param {function} [onDismiss]
+ */
+export default class TopBanner extends Component {
+  componentDidMount() {
+    Utils.waitOneTick().then(() => this.setState({ show: this.props.isVisible }), 0)
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    return _.isUndefined(state.show) ? null : { show: props.isVisible }
+  }
+
+  render() {
+    const { showX = true, children, onDismiss, ...props } = _.omit('isVisible', this.props)
+    const { show } = this.state
+
+    return div(_.merge({
+      style: {
+        display: 'flex', alignItems: 'center',
+        transition: 'all 0.25s linear',
```

Comment:
What if the transition out was instantaneous so I don't have to see the "1/0" while it slides up?

Reply:
I tried that, it looked kind of sloppy to my eye with the nice animate in.